Orange/statistics/util.py

```diff
@@ -476,6 +476,45 @@ def nanmean(x, axis=None, weights=None):
     return means
 
 
+def nan_mean_variance_axis(x, axis=None, weights=None):
+    """
+    Computes means and variance of dense and sparse matrices.
+    Supports weights. Based on mean_variance_axis.
+    """
+    if axis is None:
+        raise NotImplementedError("axis=None is not supported")
+
+    if not sp.issparse(x):
+        if weights is None:
+            means = bn.nanmean(x, axis=axis)
+            variances = bn.nanvar(x, axis=axis)
+        else:
+            if axis == 0:
+                weights = weights.reshape(-1, 1)
+            elif axis == 1:
+                weights = weights.reshape(1, -1)
+            else:
+                raise NotImplementedError
+
+            nanw = ~np.isnan(x) * weights  # do not divide by non-used weights
+            wsum = np.sum(nanw, axis=axis)
+            means = bn.nansum(x * weights, axis=axis) / wsum
+
+            if axis == 0:
+                mr = means.reshape(1, -1)
+            elif axis == 1:
+                mr = means.reshape(-1, 1)
+
+            variances = bn.nansum(((x - mr) ** 2) * weights, axis=axis) / wsum
+    else:
+        # mean_variance_axis is picky regarding the input type
+        if weights is not None:
+            weights = weights.astype(float)
+        means, variances = mean_variance_axis(x, axis=axis, weights=weights)
+
+    return means, variances
+
+
 def nanvar(x, axis=None, ddof=0):
```

Comment:
And the same for `nanstd`

Reply:
Then I'd also have to rewrite these functions, which I did not want to go into. But yes, what you suggest makes sense, I just did not need it in this story.

Also, there is `ddof`. The fix I wrote you about once, `*N/(N-ddof)`, would theoretically work, but `N` would need to be the number of non-nan elements for that particular column/row.

Maybe someone could do this in a separate PR? :)

Reply:
I think a separate PR would be fine. It seems like changing `nanvar` to include weights might be a bit of work, since it would require changing `_apply_func` as well, and that might complicate things. If we do get around to fixing it, the ddof correction would definitely be the way to go :)